src/hooks/usePaymentMethodStatus.tsx

```diff
@@ -0,0 +1,34 @@
+import { useCallback, useContext } from 'react';
+import getPaymentMethodStatus from 'actions/getPaymentMethodStatus';
+import useAsync from 'hooks/useAsync';
+import type { UseAsyncReturnType } from 'hooks/useAsync';
+import { TokenContext } from 'hooks/useCustomerToken';
+import { components } from '../apiTypes';
+
+type CustomerPaymentMethodStatus =
+  components['schemas']['CustomerPaymentMethodStatus'];
+export type UsePaymentMethodStatusReturnType =
+  UseAsyncReturnType<CustomerPaymentMethodStatus>;
+
+/**
+ * A hook that fetches customer's payment method status.
+ */
+const usePaymentMethodStatus = (args?: {
+  token?: string;
+}): UsePaymentMethodStatusReturnType => {
+  const { token: tokenFromContext } = useContext(TokenContext);
+
+  if (!args?.token && !tokenFromContext) {
+    throw new Error('Token must be provided.');
+  }
+
+  const userToken = args?.token || tokenFromContext;
```

Comment:
Minor nit: if you define `userToken` first, you could use it in that `if` block.

```suggestion
  const userToken = args?.token || tokenFromContext;

  if (!userToken) {
    throw new Error('Token must be provided.');
  }
```